cmake/kokkos_test_cxx_std.cmake

```diff
@@ -93,7 +89,7 @@ ENDIF()
 # For compiling CUDA code using nvcc_wrapper, we will use the host compiler's
 # flags for turning on C++11.  Since for compiler ID and versioning purposes
 # CMake recognizes the host compiler when calling nvcc_wrapper, this just
-# works.  Both NVCC and nvcc_wrapper only recognize '-std=c++11' which means
+# works.  Both NVCC and nvcc_wrapper only recognize '-std=c++14' which means
```

Comment:
You missed a `C++11` a couple of lines above and a `gnu++11` a couple lines below.